bitnami/thanos/templates/ruler/statefulset.yaml

```diff
@@ -142,6 +145,12 @@ spec:
               mountPath: /conf/objstore
             - name: data
               mountPath: /data
+            {{- if .Values.ruler.alertmanagersConfig }}
+            {{- if not .Values.ruler.alertmanagersConfig.file }}
```

Comment:
Could you change this lines to:
```
{{- if (and (not (empty .Values.ruler.alertmanagersConfig)) (empty .Values.ruler.alertmanagersConfig.file))}}
```

Reply:
Hmmm, doing this results in ```Error: template: thanos/templates/ruler/statefulset.yaml:148:86: executing "thanos/templates/ruler/statefulset.yaml" at <.Values.ruler.alertmanagersConfig.file>: nil pointer evaluating interface {}.file``` when just using the `alertmanagers` key. Likely because the `alertmanagers` struct isn't created.

I can check the fields in the helpers first and then just call that boolean later on -- would that work?

Reply:
Ah! I think I have a better solution.

Reply:
Yes, I did. I can likely just change `alertmanagersConfig.file` to a flat tree, as per the comment here https://github.com/helm/helm/issues/8026 and then things should work out fine.